tests/components/zwave_me/test_remove_stale_devices.py

```diff
@@ -0,0 +1,72 @@
+"""Test the zwave_me removal of stale devices."""
+from unittest.mock import patch
+
+import pytest as pytest
+from zwave_me_ws import ZWaveMeData
+
+from homeassistant.components.zwave_me import ZWaveMePlatform
+from homeassistant.const import CONF_TOKEN, CONF_URL
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry, mock_device_registry
+
+DEFAULT_DEVICE_INFO = ZWaveMeData(
+    id="DummyDevice",
+    deviceType=ZWaveMePlatform.BINARY_SENSOR,
+    title="DeviceDevice",
+    level=100,
+    deviceIdentifier="16-23",
+)
+
+
+@pytest.fixture
+def device_reg(hass):
+    """Return an empty, loaded, registry."""
+    return mock_device_registry(hass)
+
+
+async def mock_connection(controller):
+    """Mock established connection and setting identifiers."""
+    controller.on_new_device(DEFAULT_DEVICE_INFO)
+    return True
+
+
+@pytest.mark.parametrize(
+    "identifier,should_exist",
+    [
+        (DEFAULT_DEVICE_INFO.id, False),
+        (DEFAULT_DEVICE_INFO.deviceIdentifier, True),
+    ],
+)
+async def test_remove_stale_devices(
+    hass: HomeAssistant, device_reg, identifier, should_exist
+):
+    """Test removing devices with old-format ids."""
+
+    config_entry = MockConfigEntry(
+        domain="zwave_me", data={CONF_TOKEN: "test_token", CONF_URL: "http://test_test"}
+    )
```

Comment:
For clarity in the test, I'd set a `unique_id` of the MockConfigEntry, other than `None`.